lxd/auth/authorization.go

```diff
@@ -45,16 +95,38 @@ type UserAccess struct {
 	Projects map[string][]string
 }
 
-func LoadAuthorizer(name string, config map[string]any, logger logger.Logger, projectsGetFunc func() (map[int64]string, error)) (Authorizer, error) {
-	driverFunc, ok := authorizers[name]
+type Opts struct {
+	config          map[string]any
+	projectsGetFunc func(ctx context.Context) (map[int64]string, error)
+}
+
+func WithConfig(c map[string]any) func(*Opts) {
+	return func(o *Opts) {
+		o.config = c
+	}
+}
+
+func WithProjectsGetFunc(f func(ctx context.Context) (map[int64]string, error)) func(*Opts) {
+	return func(o *Opts) {
+		o.projectsGetFunc = f
+	}
+}
+
+func LoadAuthorizer(ctx context.Context, driver string, logger logger.Logger, options ...func(opts *Opts)) (Authorizer, error) {
```

Comment:
Do we need to support different combinations of options via the variadic or can it be just static arguments?